pkg/query-service/app/metrics/v4/query_builder.go

```diff
@@ -21,11 +21,6 @@ func PrepareMetricQuery(start, end int64, queryType v3.QueryType, panelType v3.P
 
 	start, end = common.AdjustedMetricTimeRange(start, end, mq.StepInterval, mq.TimeAggregation)
 
-	if mq.ShiftBy != 0 {
```

Comment:
removed this from here as it is taken care in the `runBuilderQuery` function which takes care of caching as well